war/src/main/js/pluginSetupWizardGui.js

```diff
@@ -250,6 +252,16 @@ var createPluginSetupWizard = function(appendTarget) {
 		}
 		return null;
 	};
+	var setFailureStatus = function(plugData) {
+		var plugFailIdx = failedPluginNames.indexOf(plugData.name);
+		if(/.*Fail.*/.test(plugData.installStatus)) {
+			if(plugFailIdx < 0) {
+				failedPluginNames.push(plugData.name);
+			}
+		} else if(plugFailIdx > 0) {
+			failedPluginNames = failedPluginNames.slice(plugFailIdx,1);
+		}
+	};
```

Comment:
I think some comments wrt what this function is expecting, doing etc. It's not obvious to me anyway.